OpenRA.Mods.Cnc/Traits/Attack/AttackTesla.cs

```diff
@@ -91,6 +91,13 @@ public ChargeAttack(AttackTesla attack, Target target)
 
 			public override Activity Tick(Actor self)
 			{
+				if (ChildActivity != null)
+				{
+					ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
+					if (ChildActivity != null)
+						return this;
+				}
+
 				if (IsCanceling || !attack.CanAttack(self, target))
 					return NextActivity;
```

Comment:
With the frequency of this pattern appearing now (meaning everything from the start of the method up to this line), I'm wondering whether it would make sense to bake this into the base class somehow.

Reply:
That was the idea. I plan to do that once all the activity refactors are done. That also includes replacing pseudo-childactivities by true childactitivities where possible. That way I have a better view of how any remaining special cases should be accommodated.

With this PR I just want to get the simple formulaic fixes out of the way first.